.travis.yml

```diff
@@ -33,10 +33,17 @@ install:
   # interactive terminal capabilities.
   - (cd server && stack setup -j2 --no-terminal)
   - (cd client-haskell && stack setup -j2 --no-terminal)
+  - curl -o install-nix-2.3.1 https://nixos.org/releases/nix/nix-2.3.1/install
+  - sha256sum --check nix/nix_sha256sums
+  - sh ./install-nix-2.3.1 --no-daemon
+  - . $HOME/.nix-profile/etc/profile.d/nix.sh
 
 script:
   # Build and test.
   - (cd server && stack build -j2 --no-terminal)
   - (cd server && stack test  -j2 --no-terminal)
   - (cd client-haskell && stack build -j2 --no-terminal)
   - (cd client-haskell && stack test  -j2 --no-terminal)
+  # Check that the nix builds work
+  - nix-build --no-out-link -E '(import ./nix/nixpkgs.nix {}).haskellPackages.callPackage ./client-haskell/client.nix {}'
+  - nix-build --no-out-link ./server/default.nix
```

Comment:
We could also create a `default.nix` in the repository root and use that to build both the client and the server

Reply:
I would reserve a top-level `default.nix` for proper integration in the nix-ecosystem, e.g. a nixpkg overlay providing the two packages.
But since it also has to override the version of the `websockets` library, this is best left for a separate PR.